File: src/main/java/com/vc/medium/L370.scala
package com.vc.medium

object L370 {
  def getModifiedArray(length: Int, updates: Array[Array[Int]]): Array[Int] = {
    val arr = new Array[Int](length)

    for(list <- updates) {
      val inc = list(2)
      arr(list(0)) += inc
      if(list(1) + 1 < length) arr(list(1) + 1) -= inc
    }

    var cur = 0
    (0 until length).foreach(i => {
      cur += arr(i)
      arr(i) = cur
    })
    arr
  }
}

File: src/main/java/com/vc/medium/L370.java
package com.vc.medium;

class L370 {
    public int[] getModifiedArray(int length, int[][] updates) {
        int[] arr = new int[length];

        for(int[] update: updates) {
            int startIndex = update[0];
            int endIndex = update[1];
            int inc = update[2];

            arr[startIndex] += inc;

            if(endIndex + 1 < length) arr[endIndex + 1] -= inc;
        }

        int sum = 0;
        for(int i = 0; i < length; i++) {
            sum += arr[i];
            arr[i] = sum;
        }
        return arr;
    }
}
